astropy/coordinates/tests/test_frames.py

```diff
@@ -821,11 +820,9 @@ def test_represent_as():
     assert isinstance(rep2.differentials['s'], r.CylindricalDifferential)
 
     # single class with positional in_frame_units, verify that warning raised
-    with catch_warnings() as w:
+    with pytest.warns(AstropyWarning, match='argument position') as w:
         icrs.represent_as(r.CylindricalRepresentation, False)
-        assert len(w) == 1
-        assert w[0].category == AstropyWarning
-        assert 'argument position' in str(w[0].message)
+    assert len(w) == 1
```

Comment:
I guess here there is a more explicit test that there is only one warning, which I would be tempted to remove, but fine to leave too.

Reply:
I think it is valid to ensure that is the only warning being raised, so I kept it.